static/js/components/FollowupRequestForm.jsx

```diff
@@ -0,0 +1,288 @@
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import DatePicker from 'react-datepicker';
+
+import { showNotification } from 'baselayer/components/Notifications';
+import * as Actions from '../ducks/source';
+
+import 'react-datepicker/dist/react-datepicker-cssmodules.css';
+
+
+const FollowupRequestForm = ({ source_id, action, instrumentList, instrumentObsParams, followupRequest = null, title = "Submit new follow-up request", afterSubmit = null }) => {
```

Comment:
This form could probably benefit quite a bit from React Hook Form!  Will let you investigate and see if you agree!

Reply:
Will investigate...